bootstrap/service.go

```diff
@@ -262,26 +280,99 @@ func (bs bootstrapService) UpdateConnections(ctx context.Context, token, id stri
 			return ErrThings
 		}
 	}
-	if err := bs.configs.UpdateConnections(ctx, owner, id, channels, connections); err != nil {
+	if err := bs.configs.UpdateConnections(ctx, user.GetDomainId(), id, channels, connections); err != nil {
 		return errors.Wrap(errUpdateConnections, err)
 	}
 	return nil
 }
 
+func (bs bootstrapService) listClientIDs(ctx context.Context, userID string) ([]string, error) {
+	tids, err := bs.auth.ListAllObjects(ctx, &magistrala.ListObjectsReq{
+		SubjectType: auth.UserType,
+		Subject:     userID,
+		Permission:  auth.ViewPermission,
+		ObjectType:  auth.ThingType,
+	})
+	if err != nil {
+		return nil, errors.Wrap(svcerr.ErrNotFound, err)
+	}
+	return tids.Policies, nil
+}
+
+func (bs bootstrapService) filterAllowedThingIDs(ctx context.Context, userID string, thingIDs []string) ([]string, error) {
+	var ids []string
+	tids, err := bs.auth.ListAllObjects(ctx, &magistrala.ListObjectsReq{
```

Comment:
`bs.auth.ListAllObjects` is already called in `listClientIDs` , Why it is called again here. 
Please explain the reason